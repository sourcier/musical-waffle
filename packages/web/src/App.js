import React from 'react'
import { BrowserRouter, Route, Switch } from 'react-router-dom'
import { HelmetProvider } from 'react-helmet-async'
import { Global } from '@emotion/core'
import { hot } from 'react-hot-loader/root'

import Layout from './components/ui/Layout'
import Blog from './components/routes/Blog'
import BlogPage from './components/routes/BlogPage'
// import Home from './routes/Home'
import NotFound from './components/routes/NotFound'
import Contact from './components/routes/Contact'
import ThankYou from './components/routes/Contact/ThankYou'
import { treatments } from './libs/theme'

export default hot((props) => (
  <BrowserRouter>
    <HelmetProvider>
      <Global styles={treatments.body} />
      <Layout>
        <Switch>
          {/* <Route path="/" exact component={Home} /> */}
          <Route path="/contact" exact component={Contact} />
          <Route path="/contact/thank-you" exact component={ThankYou} />
          <Route path="/" exact component={Blog} />
          <Route path="/:slug" exact component={BlogPage} />
          <Route component={NotFound} />
        </Switch>
      </Layout>
    </HelmetProvider>
  </BrowserRouter>
))
