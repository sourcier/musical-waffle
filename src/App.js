import React, { Component } from "react";
import { BrowserRouter, Route, Switch } from "react-router-dom";
import { Masthead } from "./components/Masthead";
import { Home } from "./pages/Home";
import { BlogList } from "./pages/BlogList";
import { NotFound } from "./pages/NotFound";
import { BlogPost } from "./pages/BlogPost";

class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <Masthead />
        <Switch>
          <Route exact path={"/"} component={Home} />
          <Route exact path={"/blog"} component={BlogList} />
          <Route exact path={"/blog/:slug"} component={BlogPost} />
          <Route component={NotFound} />
        </Switch>
      </BrowserRouter>
    );
  }
}

export default App;
