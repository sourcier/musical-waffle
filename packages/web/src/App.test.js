import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

jest.mock('emotion-theming', () => {
  return {
    useTheme: () => ({
      treatments: { body: {} },
    }),
  }
})

it('renders without crashing', () => {
  const component = shallow(<App restoreSession={jest.fn()} />)
  expect(component).toMatchSnapshot()
})
