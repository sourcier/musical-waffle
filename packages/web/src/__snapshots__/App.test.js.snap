// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<e>
  <HelmetProvider
    context={Object {}}
  >
    <ForwardRef
      styles={
        Object {
          "a": Object {
            "color": "#007bff",
          },
          "body": Object {
            "color": "#212529",
            "fontFamily": "'Source Sans Pro', sans-serif",
            "fontSize": "24px",
            "fontWeight": 300,
            "lineHeight": 1.5,
          },
          "button": Object {
            "cursor": "pointer",
          },
          "h1": Object {
            "fontFamily": "'Playfair Display', sans-serif",
            "fontSize": "59.712px",
            "fontWeight": 400,
            "lineHeight": 1.2057877813504823,
            "marginBottom": "36px",
          },
          "h2": Object {
            "a": Object {
              "color": "#212529",
              "fontSize": "inherit",
              "lineHeight": "inherit",
              "marginBottom": "inherit",
              "textDecoration": "none",
            },
            "fontFamily": "'Playfair Display', sans-serif",
            "fontSize": "49.775999999999996px",
            "fontWeight": 400,
            "lineHeight": 1.0848601735776278,
            "marginBottom": "36px",
          },
          "h3": Object {
            "fontFamily": "'Playfair Display', sans-serif",
            "fontSize": "41.472px",
            "fontWeight": 400,
            "lineHeight": 1.3020833333333333,
            "marginBottom": "36px",
          },
          "h4": Object {
            "fontFamily": "'Playfair Display', sans-serif",
            "fontSize": "34.56px",
            "fontWeight": 400,
            "lineHeight": 1.0416666666666665,
            "marginBottom": "36px",
          },
          "h5": Object {
            "fontFamily": "'Playfair Display', sans-serif",
            "fontSize": "28.799999999999997px",
            "fontWeight": 400,
            "lineHeight": 1.2500000000000002,
            "marginBottom": "36px",
          },
          "h6": Object {
            "fontFamily": "'Playfair Display', sans-serif",
            "fontSize": "24px",
            "fontWeight": 400,
            "lineHeight": 1.5,
            "marginBottom": "36px",
          },
          "p, a, ol, ul, pre": Object {
            "fontSize": "24px",
            "lineHeight": 1.5,
            "marginBottom": "36px",
          },
          "stong": Object {
            "fontWeight": "400",
          },
        }
      }
    />
    <Root>
      <e>
        <e
          component={[Function]}
          exact={true}
          path="/contact"
        />
        <e
          component={[Function]}
          exact={true}
          path="/contact/thank-you"
        />
        <e
          component={
            Object {
              "$$typeof": Symbol(react.memo),
              "WrappedComponent": [Function],
              "compare": null,
              "displayName": "Connect(Component)",
              "type": [Function],
            }
          }
          exact={true}
          path="/"
        />
        <e
          component={[Function]}
          exact={true}
          path="/:slug"
        />
        <e
          component={[Function]}
        />
      </e>
    </Root>
  </HelmetProvider>
</e>
`;
